### PR TITLE
Allow variable number of users in seeding.

### PR DESCRIPTION
**Pull request by RobertCram**
PT Story: https://www.pivotaltracker.com/story/show/148949923

**Changes proposed in this pull request:**

1. Allow for a variable number of users in development seeding. When needed (e.g. when working on the seeding logic), add the line `SEED_USERS=<number of users>` to the `.env` file to speed up seeding. If the `SEED_USERS` environment variable is not present, the number of users to seed reverts to the usual 100.

2. Rename `seed_spec.rb` to `seed_production_spec.rb` and add tests for development seeding in `seed_development_spec.rb`

3. Make sure all tests clean up after themselves. Change `before(:all)` methods that empty the database before starting (and thereby masking problems), to methods that just assert the database is empty.

Ready for review:
@patmbolger @weedySeaDragon

### CODE REVIEW

File: spec/db/seed_development_spec.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+  it "kommuns are in the db" do
+    expect(Kommun.all.size).to eq(290)
+  end
+
+  it "users are in the db" do
+    expect(User.all.size).to eq(seed_users)
+  end
+
+  it "adresses are in the db" do
```
Comment (patmbolger) [inline spec/db/seed_development_spec.rb]: `addresses ....`

**Review by patmbolger — APPROVED** [review]
Looks good and restructure helps a lot.

One minor spelling issue.

File: db/seeds.rb [hunk 2]
```diff
@@ -70,8 +67,10 @@
 
   puts 'Creating additional users ...'
 
+  number_of_users = (ENV['SEED_USERS'] || SEED_USERS).to_i
```
Comment (weedySeaDragon) [inline db/seeds.rb]: I'd like to prefix the variables we use with "SHF_", so I'd like this one to be `SHF_SEED_USERS`

And this needs to be added to `.env.example`  perhaps with something like this in the

 `# These are optional environment variables:` section:`

```
# SHF_SEED_USERS=25     # The number of users to create during seeding.  (Takes precedence of the default number in seeds.db)

```

**Review by weedySeaDragon — CHANGES_REQUESTED** [review]

File: db/seeds.rb (hunk 2 above)
Comment (RobertCram) [reply db/seeds.rb]: Done,  @weedySeaDragon .

**Review by weedySeaDragon — APPROVED** [review]

**Pull request merged**